packages/atomic/src/components/search/atomic-follow-up-question-list/atomic-follow-up-question-list.tsx

```diff
@@ -0,0 +1,117 @@
+import {
+  buildSearchBox,
+  getOrganizationEndpoint,
+  SearchBox,
+  SearchBoxState,
+} from '@coveo/headless';
+import {Component, h, Prop, State} from '@stencil/core';
+import {
+  FollowUpQuestionCandidate,
+  SelectFollowUpQuestionCandidatePayload,
+} from '../../../components/common/follow-up-questions/follow-up-question-list-common';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {Bindings} from '../atomic-search-interface/atomic-search-interface';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'atomic-follow-up-question-list',
+  styleUrl: 'atomic-follow-up-question-list.pcss',
+  shadow: true,
+})
+export class AtomicFollowUpQuestionList implements InitializableComponent {
+  @InitializeBindings()
+  public bindings!: Bindings;
+
+  private searchBox!: SearchBox;
+
+  private previousQuery: string = '';
+
+  /**
+   * @internal
+   * The unique identifier of the answer configuration to use to generate the answer.
+   */
+  @Prop()
+  public answerConfigurationId!: string;
+
+  @BindStateToController('searchBox', {
+    onUpdateCallbackMethod: 'onSearchBoxStateChange',
+  })
+  @State()
+  private searchBoxState!: SearchBoxState;
+
+  @State()
+  public error!: Error;
+
+  @State()
+  private candidates: FollowUpQuestionCandidate[] = [];
+
+  public initialize() {
+    this.searchBox = buildSearchBox(this.bindings.engine);
+  }
+
+  public render() {
+    return (
+      <follow-up-question-list-common
+        candidates={this.candidates}
+        onSelectCandidate={(
+          evt: CustomEvent<SelectFollowUpQuestionCandidatePayload>
+        ) => {
+          this.onCandidateSelected(evt.detail.candidate);
+        }}
+      ></follow-up-question-list-common>
+    );
+  }
+
+  private onCandidateSelected(candidate: FollowUpQuestionCandidate) {
+    this.searchBox.updateText(candidate.question);
+    this.searchBox.submit();
+  }
+
+  // @ts-expect-error: This function is used by BindStateToController.
+  private onSearchBoxStateChange() {
+    const currentQuery = this.searchBoxState?.value ?? '';
+
+    if (currentQuery === this.previousQuery) {
+      return;
+    }
+
+    this.previousQuery = currentQuery;
+
+    // The query has been updated. Clear the existing questions.
+    this.candidates = [];
+
+    if (this.answerConfigurationId && currentQuery) {
+      const {accessToken, organizationId, environment} =
+        this.bindings.engine.state.configuration;
+      const orgEndpoint = getOrganizationEndpoint(organizationId, environment);
+
+      fetch(
```

Comment:
I would've though we do not want to fetch directly from a component. But from controllers. Beside. As soon as anyone will want to create a custom component. The headless wiring wont be there and the developper will need to call the API and listen to the query himself. I feel this might no be super optimal. 

I would personnaly have taken the path of updating one of our existing controller to listen to a query, then call the endpoint with RTKquery.

Reply:
> I would've thought we do not want to fetch directly from a component. But from controllers.

I agree with you on that point for regular components. However, this specific component is experimental. Both the API and the component are subject to changes in the future. The goal is to get feedback on the feature first so we can then productize it.

If we move forward and productize the feature, then we'll implement the `fetch` call in headless using an appropriate controller.